src/v/kafka/server/handlers/configs/config_utils.h

```diff
@@ -261,6 +261,49 @@ struct segment_size_validator {
     }
 };
 
+struct replication_factor_must_be_positive {
+    std::optional<ss::sstring>
+    operator()(const cluster::replication_factor& value) {
+        if (value <= cluster::replication_factor{0}) {
+            return fmt::format("replication factor {} must be positive", value);
+        }
+        return std::nullopt;
+    }
+};
+
+struct replication_factor_must_be_odd {
+    std::optional<ss::sstring>
+    operator()(const cluster::replication_factor& value) {
+        if (value % 2 == 0) {
+            return fmt::format("replication factor {} must be odd", value);
+        }
+        return std::nullopt;
+    }
+};
+
+template<typename T, typename... ValidatorTypes>
+requires requires(const T& value, ValidatorTypes... validators) {
+    {
+        std::get<0>(std::tuple{validators...})(value)
+    } -> std::convertible_to<std::optional<ss::sstring>>;
+    (std::is_same_v<
+       decltype(std::get<0>(std::tuple{validators...})(value)),
+       decltype(validators)>
+     && ...);
+}
```

Comment:
🤯

Reply:
Ya, might be a bit overkill...i was expecting to find more applications on the config update path but didn't find much

Reply:
this is beauitful